common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -371,7 +372,12 @@ public class @K@ObjectHashMap<V> implements @K@ObjectMap<V>, Iterable<@K@ObjectM
         if (size > maxSize) {
             // Need to grow the arrays. We take care to detect integer overflow,
             // also limit array size to ArrayList.MAX_ARRAY_SIZE.
-            rehash(adjustCapacity((int) Math.min(keys.length * 2.0, Integer.MAX_VALUE - 8)));
+            int newCapacity = keys.length << 1;
+            if(newCapacity < 0) {
+                throw new IllegalStateException("Max capacity reached at size=" + size);
+            } else {
+                rehash(newCapacity);
+            }
         } else if (size == keys.length) {
             // Open addressing requires that we have at least 1 slot available. Need to refresh
             // the arrays to clear any removed elements.
```

Comment:
I actually don't think we need this else since `keys.length` must be >= maxSize.